src/content/blog/starlight-032.mdx

```diff
@@ -0,0 +1,154 @@
+---
+title: Starlight 0.32
+description: The latest Starlight features unlock new possibilities on our path to v1. Let’s meet route middleware, plugin i18n APIs, and multisite search.
+publishDate: 2025-02-17
+authors:
+  - chris
+coverImage: '/src/content/blog/_images/starlight-032/blog-post-starlight-v0.32.webp'
+socialImage: '/src/content/blog/_images/starlight-032/og-starlight-v0.32.webp'
+---
+
+**Starlight 0.32 is here! Let’s unpack what’s new as we work our way towards a v1 release.**
+
+🧘 We’ve been stretching to make Starlight extra extendable. Here’s our latest flow for more flexibility:
+
+- [**Route data middleware**](#a-new-route-data-paradigm)
+- [**New i18n APIs for plugins**](#new-i18n-apis-for-plugins)
+- [**Updated plugin hooks**](#updated-plugin-hooks)
+- [**Multisite search support**](#multisite-search-support)
+
+To upgrade an existing Starlight site, use the automated `@astrojs/upgrade` CLI tool. This will update Starlight, Astro, and any other integrations you are using:
+
+```sh
+npx @astrojs/upgrade
+```
+
+## A new route data paradigm
+
+Historically, users and plugin authors have found themselves reaching for Starlight’s [component override](https://starlight.astro.build/guides/overriding-components/) system when all they really wanted to do was tweak the data rendered by Starlight’s default components.
```

Comment:
We mention that a route middleware is a better place for some operations, but should we say, maybe at the end, why an user may want to still reach to an override?

Reply:
Made an attempt in 130e765fd to reframe this a bit so that the distinction between the two use cases is hopefully a bit clearer.

Reply:
Really like the new version personally, I think it's way more obvious why you would want to pick one or the other approach 👍